job-descriptions/software-engineer-code-exercise.md

```diff
@@ -0,0 +1,60 @@
+- At your chosen date and time, we will send you detailed instructions. You will be building a simple API server in Go that calls a JSON endpoint of another service and exposes a JSON endpoint of its own.
+- You should have a recent version of Go installed on your computer.
+- You can look up documentation on the internet while you are coding.
+- You are not prohibited from using open-source libries, but the problem is easily solved using only the Go standard library.
+- After two hours, you will email us your solution as a zip file and we will get back to you on next steps within 2 business days.
+
+If we decide to move forward, we will schedule a 30-minute followup call to discuss your code.
+
+## FAQ
+
+### Why do you do a coding exercise?
+
+A significant part of your job as a software engineer is to write code, so we naturally want to measure your ability here during our interview process.
+
+An asynchronous coding exercise provides the most realistic environment for measuring how you would work at Sourcegraph on a daily basis.
+
+- You can use the programming environment that you are most familiar with (i.e. your own laptop with your own editor with all the keyboard shortcuts and extensions/tools that you like, and with access to documentation on the internet).
+- You don't have the extra pressure of coding while someone else is watching.
+- You can schedule our asynchronous coding exercise for the best time that works for you.
+
+Candidates who have completed our coding exercise regularly tell us that they enjoyed it and/or learned something.
+
+### Why a 2-hour limit?
+
+We choose to enforce a finite limit so that we get a consistent signal across all candidates. It wouldn't be fair if candidates could spend as much time as they wanted, because some candidates have more time than others.
+
+2 hours is long enough that we can ask non-trivial questions that test for relevant skills while also being short enough that it shouldn't be an unreasonable burden for anyone.
+
+### What if I have significant open-source contributions?
+
+Many engineers on our team have made significant open-source contributions, so we view that as a strong positive signal on your application and we will probably ask you about your contributions during interviews; however, we still ask all candidates to complete our coding exercise.
+
+Our coding exercise provides a consistent and fair signal across all candidates for skill that we care about. If we just review your open-source code, we have a hard time comparing you to other candidates fairly and we might not get signal on specific skills that we care about.
+
+Even if it is obvious that the coding exercise is a formality for you (e.g. maybe you are a language author), we would still ask you to do it because we would want your feedback on the exercise so we can make it better for future candidates.
```

Comment:
Trivial nit.

```suggestion
Even if it is obvious that the coding exercise is a formality for you (e.g., maybe you are a language author), we would still ask you to do it because we would want your feedback on the exercise so we can make it better for future candidates.
```